nncf/pruning/base_algo.py

```diff
@@ -60,95 +70,202 @@ def apply_to(self, target_model: NNCFNetwork) -> NNCFNetwork:
+                cluster = NodesCluster(i, list(all_pruned_inputs), [n.id for n in all_pruned_inputs])
+                pruned_nodes_clusterization.add_cluster(cluster, i)
+
+        last_cluster_idx = len(special_ops_clusterization.get_all_clusters())
+
+        # 3. Add remaining single nodes as separate clusters
+        for node in all_nodes_to_prune:
+            if not pruned_nodes_clusterization.is_not_in_any_cluster(node.node_id):
+                scope = node.op_exec_context.scope_in_model
+                module = target_model.get_module_by_scope(scope)
+                node_info = NodeInfo(node, module, scope)
+
+                cluster = NodesCluster(last_cluster_idx, [node_info], [node.node_id])
+                pruned_nodes_clusterization.add_cluster(cluster, last_cluster_idx)
+
+                last_cluster_idx += 1
+
+        # 4. Merge clusters for Conv + Depthwise conv (should be pruned together too)
+        for node in all_nodes_to_prune:
+            scope = node.op_exec_context.scope_in_model
+            module = target_model.get_module_by_scope(scope)
+            cluster_id = pruned_nodes_clusterization.get_cluster_for_node(node.node_id).id
+
+            if is_depthwise_conv(module):
+                previous_conv = get_previous_conv(target_model, module, scope)
+                if previous_conv:
+                    previous_conv_cluster_id = pruned_nodes_clusterization.get_cluster_for_node(
+                        previous_conv.node_id).id
+                    pruned_nodes_clusterization.merge_clusters(cluster_id, previous_conv_cluster_id)
+
+        # 5. Checks for groups (all nodes in group can be pruned or all group can't be pruned).
+        model_analyser = ModelAnalyser(target_model)
+        can_prune_analysis = model_analyser.analyse_model_before_pruning()
+        self._check_pruned_groups(target_model, pruned_nodes_clusterization, can_prune_analysis)
+        return pruned_nodes_clusterization
+
+    def _should_prune_module_check(self, target_model: NNCFNetwork, module, module_scope: Scope):
+        """
+        Check whether we should prune module according to algorithm parameters.
+        :param target_model: model to work with
+        :param module: module to check
+        :param module_scope: scope for module
+        :return: (prune: bool, msg: str)
+        prune: Whether we should prune module
+        msg: additional information why we should/shouldn't prune
+        """
+        prune = True
+        msg = None
 
         input_non_pruned_modules = get_first_pruned_modules(target_model,
                                                             self.get_types_of_pruned_modules() + ['linear'])
         output_non_pruned_modules = get_last_pruned_modules(target_model,
                                                             self.get_types_of_pruned_modules() + ['linear'])
+        module_scope_str = str(module_scope)
+
+        if not self._should_consider_scope(module_scope_str):
```

Comment:
Did you remove a check for frozen layers via the `ignore_frozen_layers` flag?

Reply:
No, it was of the problem of merging with the last develop. Fixed it.